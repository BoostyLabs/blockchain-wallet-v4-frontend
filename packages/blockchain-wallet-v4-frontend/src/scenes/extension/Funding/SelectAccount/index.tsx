import React, { useState } from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

import { Text } from 'blockchain-info-components'

import CheckBox from '../../../../icons/BackIcon'

const List = styled.ul`
  width: 100%;
  margin: 60px 0 0;
  padding: 0;
  list-style: none;
  .item {
    margin-top: 30px;
    padding: 0;
    display: flex;
    align-items: center;
    background: none;
    border: none;
  }
`
export const ListItem = styled.div`
  margin-left: 16px;
  width: 100%;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(2, 1fr);
  font-size: 16px;
  line-height: 24px;
`
const ButtonWrapper = styled.div`
  width: 100%;
  flex-grow: 1;
  display: flex;
  align-items: end;
`

export const Continue = styled(Link)`
  width: 100%;
  padding: 16px;
  background: #65a5ff;
  border-radius: 8px;
  color: #0e121b;
  text-align: center;
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  text-decoration: none;
  cursor: pointer;
`

export const FundingHeading = styled(Text)`
  font-size: 20px;
  margin: 34px 0 67px;
  color: white;
`

export const SelectAccount = () => {
  const [selectedAccount, setSelectedAccount] = useState<string>('Trading')

  return (
    <>
      <FundingHeading style={{ marginBottom: 0 }}>Add crypto</FundingHeading>
      <Text size='14px' lineHeight='20px' style={{ color: '#98A1B2', marginTop: '12px' }}>
        You can transfer crypto from your Blockchain account or another wallet
        <Link to='/' style={{ display: 'block' }}>
          Learn more
        </Link>
      </Text>
      <List>
        <li style={{ cursor: 'pointer' }}>
          <button type='button' className='item' onClick={() => setSelectedAccount('Trading')}>
            <CheckBox
              innerColor={selectedAccount ? '#619FF7' : 'transparent'}
              outerColor={selectedAccount ? '#619FF7' : '#98A1B2'}
            />
            <ListItem>
              <Text color='white'>Trading Account</Text>
              <Text color='white' style={{ textAlign: 'right' }}>
                $3,225.01
              </Text>
              <Text size='14px' lineHeight='20px' style={{ textAlign: 'left' }}>
                Ethereum
              </Text>
              <Text size='14px' lineHeight='20px' style={{ textAlign: 'right' }}>
                5.3655 ETH
              </Text>
            </ListItem>
          </button>
        </li>
        <li style={{ cursor: 'pointer' }}>
          <button type='button' className='item' onClick={() => setSelectedAccount('')}>
            <CheckBox
              innerColor={selectedAccount ? 'transparent' : '#619FF7'}
              outerColor={selectedAccount ? '#98A1B2' : ' #619FF7'}
            />
            <Text color='white' style={{ marginLeft: '16px' }}>
              Another wallet
            </Text>
          </button>
        </li>
      </List>
      <ButtonWrapper>
        <Continue to={`/extension/funding/${selectedAccount ? 'amount' : 'receive'}`}>
          Continue
        </Continue>
      </ButtonWrapper>
    </>
  )
}
