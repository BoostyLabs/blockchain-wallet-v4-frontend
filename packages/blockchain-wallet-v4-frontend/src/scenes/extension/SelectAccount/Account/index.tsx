import React, { MouseEvent } from 'react'
import { IconCheckCircle, IconCopy } from '@blockchain-com/icons'
import styled from 'styled-components'

import { Text } from 'blockchain-info-components'

import IconBitcoin from '../../../../icons/IconBitcoin'
import IconEthereum from '../../../../icons/IconEthereum'
import { SwapAccountType } from '..'

export const AccountBlock = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 16px 0;
  cursor: pointer;
`
const AccountInfo = styled.div`
  display: flex;
  flex-direction: column;
  margin: auto auto auto 16px;
`
const WalletBlock = styled.div`
  display: flex;
  align-items: center;
  cursor: copy;
`
type AccountProps = {
  account: SwapAccountType
  activeAccountIndex: number
  setActiveAccountIndex: (activeAccountIndex: number) => void
}

const Account: React.FC<AccountProps> = ({
  account,
  activeAccountIndex,
  setActiveAccountIndex
}) => {
  const copyAddress = (event: MouseEvent<HTMLDivElement>, address: string | number): void => {
    event.stopPropagation()
    navigator.clipboard.writeText(address.toString())
    // TODO: Some notification about copy address.
  }

  const shortingAddress = (address: string | number): string => {
    const addressString: string = address.toString()
    return `${addressString.slice(0, 4)}...${addressString.slice(-4)}`
  }

  return (
    <AccountBlock
      key={account.accountIndex}
      onClick={() => setActiveAccountIndex(account.accountIndex)}
      className='account'
    >
      {account.coin === 'ETH' ? <IconEthereum /> : <IconBitcoin />}
      <AccountInfo>
        <Text size='12px' weight={500} style={{ color: '#98A1B2', lineHeight: '150%' }}>
          {account.label}
        </Text>
        <WalletBlock
          onClick={(event: MouseEvent<HTMLDivElement>) => copyAddress(event, account.address)}
          className='wallet-address'
        >
          <Text size='14px' color='white' weight={500} style={{ lineHeight: '150%' }}>
            {shortingAddress(account.address)}
          </Text>
          <IconCopy color='#ffffff' height='12px' width='12px' style={{ paddingLeft: '5px' }} />
        </WalletBlock>
        <Text size='14px' color='white' weight={500} style={{ lineHeight: '150%' }}>
          {account.balance} {account.coin}
        </Text>
        <Text size='12px' weight={500} style={{ color: '#98A1B2', lineHeight: '150%' }}>
          $5,225.01
        </Text>
      </AccountInfo>
      {activeAccountIndex === account.accountIndex && (
        <IconCheckCircle height='24px' width='24px' style={{ fill: '#65A5FF' }} />
      )}
    </AccountBlock>
  )
}

export default Account
