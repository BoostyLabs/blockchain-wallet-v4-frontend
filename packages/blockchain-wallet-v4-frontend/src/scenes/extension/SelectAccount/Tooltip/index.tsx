import React from 'react'
import styled from 'styled-components'

import { Text } from 'blockchain-info-components'

const TooltipText = styled(Text)`
  white-space: nowrap;
`

interface TooltipProps {
  backgroundColor: string
  index: number
  text: string
  textColor: string
}

const Tooltip: React.FC<{ copyTooltipProperties: TooltipProps }> = ({ copyTooltipProperties }) => {
  const TooltipWrapper = styled.div`
    position: absolute;
    top: -35px;
    left: 50%;
    transform: translate(-50%, 0);
    border-radius: 4px;
    padding: 6px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    background: ${copyTooltipProperties.backgroundColor};
    &:before {
      content: '';
      position: absolute;
      top: 30px;
      left: 50%;
      transform: translate(-50%, 0);
      width: 0;
      height: 0;
      border-left: 6px solid transparent;
      border-right: 6px solid transparent;
      border-top: 6px solid ${copyTooltipProperties.backgroundColor};
    }
  `
  return (
    <TooltipWrapper id='tooltip'>
      <TooltipText
        size='12px'
        lineHeight='150%'
        weight={500}
        color={copyTooltipProperties.textColor}
      >
        {copyTooltipProperties.text}
      </TooltipText>
    </TooltipWrapper>
  )
}

export default Tooltip
