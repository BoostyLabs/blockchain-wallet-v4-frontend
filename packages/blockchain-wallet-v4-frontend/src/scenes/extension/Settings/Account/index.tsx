import React from 'react'
import { Icon } from '@blockchain-com/constellation'
import { IconChevronRightV2 } from '@blockchain-com/icons'
import styled from 'styled-components'

import { Setting, SettingsHeading } from '..'
import { SettingsLink, SettingsLinkLabel, SettingsList } from '../Overal'

const AccountSettingLabel = styled(SettingsLinkLabel)`
  margin-left: 0;
`

export const Account = (props) => {
  const { path } = props.match

  const accountSettings = [
    new Setting('Import account', `${path}/import`),
    new Setting('Upgrade account', `${path}/upgrade`)
  ]
  return (
    <>
      <SettingsHeading>Account</SettingsHeading>
      <SettingsList>
        {accountSettings.map((setting: Setting) => (
          <li key={setting.label}>
            <SettingsLink to={setting.path}>
              <AccountSettingLabel>{setting.label}</AccountSettingLabel>
              <Icon color='white800' label='IconBack' size='md'>
                <IconChevronRightV2 />
              </Icon>
            </SettingsLink>
          </li>
        ))}
      </SettingsList>
    </>
  )
}
