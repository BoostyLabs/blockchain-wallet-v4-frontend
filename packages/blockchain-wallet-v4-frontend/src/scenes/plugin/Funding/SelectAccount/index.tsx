import React, { useState } from 'react'
import { FormattedMessage } from 'react-intl'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { Icon } from '@blockchain-com/constellation'
import { IconCheckCircle } from '@blockchain-com/icons'
import { bindActionCreators } from 'redux'
import styled from 'styled-components'

import { Text } from 'blockchain-info-components'
import FiatDisplay from 'components/Display/FiatDisplay'
import { actions } from 'data'

const List = styled.ul`
  width: 100%;
  margin: 60px 0 0;
  padding: 0;
  list-style: none;
  .item {
    margin-top: 30px;
    padding: 0;
    display: flex;
    align-items: center;
    background: none;
    border: none;
  }
`
export const ListItemContent = styled.div`
  margin-left: 16px;
  width: 100%;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(2, 1fr);
  font-size: 16px;
  line-height: 24px;
`
const ButtonWrapper = styled.div`
  width: 100%;
  flex-grow: 1;
  display: flex;
  align-items: end;
`

export const Continue = styled(Link)`
  width: 100%;
  padding: 16px;
  background: ${(props) => props.theme.white};
  border-radius: 8px;
  color: ${(props) => props.theme.exchangeLogin};
  text-align: center;
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  text-decoration: none;
  cursor: pointer;
`

export const FundingHeading = styled(Text)`
  font-size: 20px;
  margin: 34px 0 67px;
  color: ${(props) => props.theme.white};
`
export const Subtitle = styled(Text)`
  font-size: 14px;
  line-height: 20px;
  color: ${(props) => props.theme.grey400};
`

const Balance = styled(Text)`
  display: 'flex';
  justify-content: 'right';
`

const LeftAlignedText = styled(Text)`
  text-align: left;
`
const RightAlignedText = styled(Text)`
  text-align: right;
`

const ListElement = styled.li`
  cursor: pointer;
`
const AnotherWalletLabel = styled(Text)`
  margin-left: 16px;
`
const SelectAccount = () => {
  const [selectedAccount, setSelectedAccount] = useState<string>('Trading')

  return (
    <>
      <FundingHeading style={{ marginBottom: '12px' }}>
        <FormattedMessage id='scenes.plugin.funding.add_crypto' defaultMessage='Add Crypto' />
      </FundingHeading>
      <Subtitle>
        <FormattedMessage
          id='scenes.plugin.funding.description'
          defaultMessage='You can transfer crypto from your Blockchain account or another wallet'
        />
        <br />
        <Link to='/'>
          <FormattedMessage id='buttons.learn_more' defaultMessage='Learn more' />
        </Link>
      </Subtitle>
      <List>
        <ListElement>
          <button type='button' className='item' onClick={() => setSelectedAccount('Trading')}>
            <Icon color='white800' label='IconCheckCircle' size='md'>
              <IconCheckCircle />
            </Icon>
            <ListItemContent>
              <Text color='white'>
                <FormattedMessage
                  id='scenes.debit_card.dashboard.funds.type.trading_account'
                  defaultMessage='Trading Account'
                />
              </Text>
              <Balance color='white'>
                <FiatDisplay color='grey400' size='12px' weight={500} coin='ETH' />
              </Balance>
              <LeftAlignedText size='14px' lineHeight='20px'>
                <FormattedMessage
                  id='scenes.settings.addresses.menutop.eth'
                  defaultMessage='Ethereum'
                />
              </LeftAlignedText>
              <RightAlignedText size='14px' lineHeight='20px' />
            </ListItemContent>
          </button>
        </ListElement>
        <li style={{ cursor: 'pointer' }}>
          <button type='button' className='item' onClick={() => setSelectedAccount('')}>
            <Icon color='white800' label='IconCheckCircle' size='md'>
              <IconCheckCircle />
            </Icon>
            <AnotherWalletLabel color='white'>
              <FormattedMessage
                id='scenes.plugin.funding.another_wallet'
                defaultMessage='Another wallet'
              />
            </AnotherWalletLabel>
          </button>
        </li>
      </List>
      <ButtonWrapper>
        <Continue to={`/extension/funding/${selectedAccount ? 'amount' : 'receive'}`}>
          <FormattedMessage id='buttons.continue' defaultMessage='Continue' />
        </Continue>
      </ButtonWrapper>
    </>
  )
}

export default SelectAccount
