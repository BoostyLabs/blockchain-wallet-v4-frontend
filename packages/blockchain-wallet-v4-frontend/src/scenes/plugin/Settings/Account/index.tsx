import React from 'react'
import { FormattedMessage } from 'react-intl'
import { Icon } from '@blockchain-com/constellation'
import { IconChevronRightV2 } from '@blockchain-com/icons'
import styled from 'styled-components'

import { Flex } from 'components/Flex'

import { Setting, SettingsHeading } from '..'
import { SettingsLink, SettingsLinkLabel, SettingsList } from '../Overal'

const AccountSettingLabel = styled(SettingsLinkLabel)`
  margin-left: 0;
`

export const Account = (props) => {
  const { path } = props.match

  const accountSettings = [
    new Setting('', 'Import account', `${path}/import`),
    new Setting('', 'Upgrade account', `${path}/upgrade`)
  ]
  return (
    <>
      <SettingsHeading>
        <FormattedMessage id='scenes.plugin.settings.account.heading' defaultMessage='Account' />
      </SettingsHeading>
      <SettingsList>
        {accountSettings.map((setting: Setting) => (
          <li key={setting.label}>
            <SettingsLink to={setting.path}>
              <Flex justifyContent='space-between' alignItems='center'>
                <AccountSettingLabel>{setting.label}</AccountSettingLabel>
                <Icon color='white800' label='IconBack' size='md'>
                  <IconChevronRightV2 />
                </Icon>
              </Flex>
            </SettingsLink>
          </li>
        ))}
      </SettingsList>
    </>
  )
}
