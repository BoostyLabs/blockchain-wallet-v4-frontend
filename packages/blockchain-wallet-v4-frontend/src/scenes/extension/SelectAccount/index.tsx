import React, { useState } from 'react'
import { connect, useSelector } from 'react-redux'
import { IconClose } from '@blockchain-com/icons'
import styled from 'styled-components'

import { CoinType } from '@core/types'
import { Button, Text } from 'blockchain-info-components'
import { getTotalBalance } from 'components/Balances/total/selectors'
import { SwapBaseCounterTypes } from 'data/components/swap/types'
import { RootState } from 'data/rootReducer'

import { Account } from './Account'

const Wrapper = styled.div`
  padding: 27px 24px;
  background: ${(props) => props.theme.exchangeLogin};
`
const IconWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  cursor: pointer;
`
const SubHeaderText = styled(Text)`
  margin: 0px 0 36px;
  color: #98a1b2;
`
const HeaderText = styled(Text)`
  margin: 30px 0 12px;
`
const ImportButton = styled(Button)`
  background-color: ${(props) => props.theme.exchangeLogin};
  border: 1px solid #0c6cf2;
  margin-top: 158px;
  color: #0c6cf2;
`

// TODO: Mock SwapAccount entity.
export class SwapAccountType {
  // eslint-disable-next-line no-useless-constructor
  public constructor(
    public balance: number | string = '',
    public baseCoin: string,
    public coin: CoinType,
    public label: string,
    public type: SwapBaseCounterTypes,
    public accountIndex: number,
    public address: number | string,
    public archived?: boolean,
    public index?: number
  ) {}
}

const SelectAccount = (props) => {
  const [activeAccountIndex, setActiveAccountIndex] = useState<number>(0)
  const [copiedWalletAddress, setCopiedWalletAddress] = useState<string | number>('')
  // @ts-ignore
  const address = useSelector((state: RootState) => state.dataPath.eth.addresses.data)

  const {
    data: { totalBalance }
  } = props

  // TODO: Mock accounts data.
  const accounts: SwapAccountType[] = [
    new SwapAccountType(
      totalBalance,
      'Ethereum',
      'ETH',
      'Ethereum account',
      SwapBaseCounterTypes.CUSTODIAL,
      0,
      Object.keys(address)[0],
      false,
      0
    )
  ]

  return (
    <Wrapper>
      <IconWrapper>
        <IconClose color='#ffffff' height='24px' width='24px' />
      </IconWrapper>
      <HeaderText size='20px' color='white' weight={500}>
        Select account
      </HeaderText>
      <SubHeaderText size='14px' weight={500}>
        {`Total Balance ${totalBalance}`}
      </SubHeaderText>
      {accounts.length &&
        accounts.map((account: SwapAccountType) => (
          <Account
            key={account.accountIndex}
            account={account}
            setActiveAccountIndex={setActiveAccountIndex}
            activeAccountIndex={activeAccountIndex}
            setCopiedWalletAddress={setCopiedWalletAddress}
            copiedWalletAddress={copiedWalletAddress}
          />
        ))}
      <ImportButton capitalize data-e2e='s' height='48px' fullwidth size='16px'>
        Add account
      </ImportButton>
    </Wrapper>
  )
}

const mapStateToProps = (state) => {
  const data = getTotalBalance(state)
  return data
}

export default connect(mapStateToProps, null)(SelectAccount)
