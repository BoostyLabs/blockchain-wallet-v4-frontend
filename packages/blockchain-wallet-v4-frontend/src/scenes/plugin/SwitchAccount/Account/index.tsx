import React, { MouseEvent, useEffect, useState } from 'react'
import { FormattedMessage } from 'react-intl'
import { connect, useDispatch, useSelector } from 'react-redux'
import { IconCheckCircle, IconCopy } from '@blockchain-com/icons'
import { SwapAccountType } from 'blockchain-wallet-v4-frontend/src/data/components/swap/types'
import Tooltip from 'blockchain-wallet-v4-frontend/src/scenes/plugin/SwitchAccount/Tooltip'
import { bindActionCreators } from 'redux'
import styled from 'styled-components'

import { Icon, Text } from 'blockchain-info-components'
import CoinDisplay from 'components/Display/CoinDisplay'
import FiatDisplay from 'components/Display/FiatDisplay'
import { Flex } from 'components/Flex'
import { actions, selectors } from 'data'
import { actions as cacheActions } from 'data/cache/slice'
import { RootState } from 'data/rootReducer'

const AccountBlock = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 22px 0;
  cursor: pointer;
`
const ConnectBlock = styled.div`
  position: relative;
  #tooltip {
    display: none;
  }
  &:hover {
    #tooltip {
      display: block;
    }
  }
`
const AccountInfo = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  margin: auto auto auto 16px;
  #tooltip {
    display: none;
  }
  &:hover {
    #tooltip {
      display: block;
    }
  }
`
const IconCheckWrapper = styled(IconCheckCircle)`
  path {
    fill: ${(props) => props.theme.white};
  }
`
const IconUncheckCircle = styled.div`
  width: 24px;
  height: 24px;
  border-radius: 50%;
  border: 2px solid ${(props) => props.theme.grey800};
`
const WalletBlock = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  cursor: copy;
`
const CoinTextBlock = styled(Text)`
  display: flex;
`

interface TooltipProperties {
  backgroundColor: string
  index: number
  leftBlockPosition: number
  leftTrianglePosition: number
  text: string
  textColor: string
}

type AccountProps = {
  account: SwapAccountType[]
  copiedAccountIndex: number
  index: number
  selectedAccountIndex: number
  setCopiedAccountIndex: (copiedAccountIndex: number) => void
  setSelectedAccountIndex: (selectedAccountIndex: number) => void
}

const defaultTooltipProperties: TooltipProperties = {
  backgroundColor: 'black',
  index: 0,
  leftBlockPosition: 50,
  leftTrianglePosition: 50,
  text: 'Copy to clipboard',
  textColor: 'grey400'
}

const Account: React.FC<AccountProps & any> = ({
  account,
  addressR,
  copiedAccountIndex,
  index,
  setCopiedAccountIndex,
  setIsSwitchAccountVisible,
  ...props
}) => {
  const dispatch = useDispatch()
  const selectedAccount = useSelector((state) => selectors.cache.getCache(state).selectedAccount)
  const state = useSelector((state: RootState) => state)

  const balance = selectors.balances.getCoinTotalBalance(account[0].coin)(state).getOrElse(0)

  const [tooltipProperties, setTooltipProperties] =
    useState<TooltipProperties>(defaultTooltipProperties)

  useEffect(() => {
    props.requestActions.getNextAddress(account[0])
  }, [])

  const changeTooltipProperties = (
    backgroundColor: string,
    index: number,
    leftBlockPosition: number,
    leftTrianglePosition: number,
    text: string,
    textColor: string
  ): TooltipProperties => {
    return { backgroundColor, index, leftBlockPosition, leftTrianglePosition, text, textColor }
  }

  const generateNextAddress = () => {
    return addressR.cata({
      Failure: () =>
        setTooltipProperties(
          changeTooltipProperties('white', index, 50, 50, 'Please try again', 'black')
        ),
      Loading: () =>
        setTooltipProperties(
          changeTooltipProperties('white', index, 50, 50, 'Getting address...', 'black')
        ),
      NotAsked: () =>
        setTooltipProperties(
          changeTooltipProperties('white', index, 50, 50, 'Please try again', 'black')
        ),
      Success: (val) => val.address
    })
  }

  const copyAddress = (event: MouseEvent<HTMLDivElement>) => {
    event.stopPropagation()
    setCopiedAccountIndex(index)
    const accountAddress = generateNextAddress()

    if (accountAddress) {
      navigator.clipboard.writeText(accountAddress)
      setTooltipProperties(changeTooltipProperties('white', index, 50, 50, 'Copied!', 'black'))
    }
  }

  const selectAccount = (account) => {
    dispatch(cacheActions.setSelectedAccount(account))
    setIsSwitchAccountVisible(false)
  }

  return (
    <AccountBlock key={index} onClick={() => selectAccount(account)}>
      <Icon size='24' name={account[0].coin} />
      <AccountInfo>
        {copiedAccountIndex === index ? (
          <Tooltip tooltipProperties={tooltipProperties} />
        ) : (
          <Tooltip tooltipProperties={defaultTooltipProperties} />
        )}
        <WalletBlock onClick={(event: MouseEvent<HTMLDivElement>) => copyAddress(event)}>
          <Text
            size='16px'
            color='white'
            weight={600}
            lineHeight='150%'
            style={{ marginRight: '9px' }}
          >
            {account.label ? (
              <FormattedMessage
                id={`plugin.switch.account.label.${account[0].label}`}
                defaultMessage={account[0].label}
              />
            ) : (
              <FormattedMessage
                id='plugin.switch.account.private_key_wallet'
                defaultMessage='Private Key Wallet'
              />
            )}
          </Text>
          <IconCopy width={16} height={16} />
        </WalletBlock>
        <Flex>
          <FiatDisplay color='grey400' size='12px' weight={500} coin={account[0].coin}>
            {balance}
          </FiatDisplay>
          <CoinTextBlock color='grey400' size='12px' weight={500}>
            &nbsp;(
            <CoinDisplay coin={account[0].coin} color='grey400' size='12px' weight={500}>
              {balance}
            </CoinDisplay>
            )
          </CoinTextBlock>
        </Flex>
      </AccountInfo>
      {selectedAccount && selectedAccount[0].coin === account[0].coin ? (
        <ConnectBlock>
          <IconCheckWrapper height='24px' width='24px' />
          <Tooltip
            tooltipProperties={changeTooltipProperties(
              'white',
              index,
              70,
              65,
              'Connected!',
              'black'
            )}
          />
        </ConnectBlock>
      ) : (
        <ConnectBlock>
          <IconUncheckCircle />
          <Tooltip
            tooltipProperties={changeTooltipProperties(
              'black',
              index,
              50,
              50,
              'Connect',
              'grey400'
            )}
          />
        </ConnectBlock>
      )}
    </AccountBlock>
  )
}

const mapStateToProps = (state, props) => ({
  addressR: selectors.components.request.getNextAddress(state, props.account[0])
})

const mapDispatchToProps = (dispatch) => ({
  requestActions: bindActionCreators(actions.components.request, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(Account)
