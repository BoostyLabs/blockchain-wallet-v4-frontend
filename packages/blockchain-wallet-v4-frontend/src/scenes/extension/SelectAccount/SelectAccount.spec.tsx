import React from 'react'
import { IconClose } from '@blockchain-com/icons'
import { shallow } from 'enzyme'

import SelectAccount from '.'
import Account from './Account'

describe('SelectAccount', () => {
  const selectAccount = shallow(<SelectAccount />)
  it('renders correctly', (): void => {
    expect(selectAccount).toMatchSnapshot()
  })

  it('Should display the "Select account" title', () => {
    expect(selectAccount.contains('Select account')).toBe(true)
  })

  it('Should display the "Import account" button', () => {
    expect(selectAccount.contains('Import account')).toBe(true)
  })

  it('Should display the one Close icon', () => {
    expect(selectAccount.find(IconClose)).toHaveLength(1)
  })

  it('Should display the account block', () => {
    expect(selectAccount.find(Account)).toBeTruthy()
  })
})
