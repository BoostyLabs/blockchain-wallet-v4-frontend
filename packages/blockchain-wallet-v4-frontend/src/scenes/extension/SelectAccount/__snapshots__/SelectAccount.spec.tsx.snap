// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectAccount renders correctly 1`] = `
<styled.div>
  <styled.div>
    <SvgIconClose
      color="#ffffff"
      height="24px"
      width="24px"
    />
  </styled.div>
  <Text
    altFont={false}
    capitalize={false}
    color="white"
    cursor="inherit"
    italic={false}
    lineHeight="inherit"
    opacity={1}
    size="20px"
    style={
      Object {
        "marginBottom": "36px",
        "marginTop": "30px",
      }
    }
    uppercase={false}
    weight={500}
  >
    Select account
  </Text>
  <Account
    account={
      SwapAccountType {
        "accountIndex": 0,
        "address": "0xb0106c26a6CfbAFB372e317f1a535304F69D3968",
        "archived": false,
        "balance": "5.20018653",
        "baseCoin": "Ethereum",
        "coin": "ETH",
        "index": 0,
        "label": "Ethereum account",
        "type": "CUSTODIAL",
      }
    }
    activeAccountIndex={0}
    key="0"
    setActiveAccountIndex={[Function]}
  />
  <Account
    account={
      SwapAccountType {
        "accountIndex": 1,
        "address": "0xb0106c26a6CfbAFB372e317f1a535304F69D3968",
        "archived": false,
        "balance": "0.20018653",
        "baseCoin": "Bitcoin",
        "coin": "BTC",
        "index": 0,
        "label": "Bitcoin account",
        "type": "CUSTODIAL",
      }
    }
    activeAccountIndex={0}
    key="1"
    setActiveAccountIndex={[Function]}
  />
  <Button
    bold={false}
    capitalize={true}
    color="#0E121B"
    data-e2e="s"
    disabled={false}
    fullwidth={true}
    height="48px"
    nature="empty"
    rounded={false}
    size="16px"
    small={false}
    style={
      Object {
        "backgroundColor": "#65A5FF",
        "border": "none",
        "marginTop": "158px",
      }
    }
    uppercase={false}
  >
    Import account
  </Button>
</styled.div>
`;
